### PR TITLE
Rework oracle actor communication

### PR DESCRIPTION
**Pull request by da-kami**

### CODE REVIEW

File: daemon/src/taker_cfd.rs [hunk 1]
```diff
@@ -271,13 +269,11 @@ impl Actor {
...
-            .clone()
-            .context("No oracle announcements available")?;
-        let offer_announcement = offer_announcements
-            .get(&cfd.order.oracle_event_id)
-            .context("Order's announcement not found in current oracle announcements")?;
+        let offer_announcement = self
+            .oracle_actor
+            .send(oracle::GetAnnouncement(cfd.order.oracle_event_id.clone()))
+            .await?
+            .context("Announcement not found")?;
```
Comment (thomaseizinger) [inline daemon/src/taker_cfd.rs]: Perhaps we should do `with_context` here and print out the actual ID.

**Review by thomaseizinger — APPROVED** [review]

**Comment by da-kami** [comment]
@bonomat do we want to have a `8` hour or `24` hour `term`?
With the changes we introduce the oracle actor still keeps a 24 hour window, but that is not exposed to the cfd actor. The cfd actor can decide to use a shorter term.

**Review by thomaseizinger — APPROVED** [review]

File: daemon/src/oracle.rs [hunk 2]
```diff
@@ -69,7 +68,7 @@ where
         }
 
         Self {
-            latest_announcements: None,
+            latest_announcements: HashMap::new(),
```
Comment (luckysori) [inline daemon/src/oracle.rs:68]: :question: Is it still worth regularly caching latest announcements now that we are not pushing them out to the CFD actor? We could simply go to the oracle with the `event_id`(s) that we constructed/received?

**Review by luckysori — APPROVED** [review]
Makes sense, nice.

File: daemon/src/oracle.rs (hunk 2 above)
Comment (thomaseizinger) [reply daemon/src/oracle.rs:68]: The idea was to decouple IO errors from the actual contract-setup.

Comment (luckysori) [reply daemon/src/oracle.rs:68]: Okay, but keep in mind that this results in making 24 HTTP requests to `olivia` every minute as opposed to 1 per CFD.

If we want caching perhaps we should only update this internal state every hour and load a wider range of events to cover our bases.

**Comment by da-kami** [comment]
added commit to cleanup/fix URL and event_id handling

**Review by thomaseizinger — APPROVED** [review]

**Comment by bonomat** [comment]
> @bonomat do we want to have a `8` hour or `24` hour `term`? With the changes we introduce the oracle actor still keeps a 24 hour window, but that is not exposed to the cfd actor. The cfd actor can decide to use a shorter term.

24 for now.

**Pull request merged**